Cannot run build task anymore

Steps to Reproduce:

1.  I was on a branch and bumped `jschardet` to `3.1.4` in all places where used
2. I ran `npm i` and the default build task
3. I switched back to `main`
4. I ran `npm i`, even `git clean -xfd` and `npm i` but I get this now:

```
Executing task in folder vscode: npm run watch-clientd 


> code-oss-dev@1.95.0 watch-clientd
> deemon npm run watch-client

[deemon] Spawned build daemon. Press Ctrl-C to detach, Ctrl-D to kill.

> code-oss-dev@1.95.0 watch-client
> node --max-old-space-size=8192 ./node_modules/gulp/bin/gulp.js watch-client

Error: Command failed: npm ls --all --omit=dev --parseable
npm ERR! code ELSPROBLEMS
npm ERR! invalid: jschardet@3.1.4 /Users/bpasero/Development/Microsoft/vscode/node_modules/jschardet

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/bpasero/.npm/_logs/2024-10-01T06_04_42_869Z-debug-0.log

    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at checkExecSyncError (node:child_process:891:11)
    at Object.execSync (node:child_process:963:15)
    at getNpmProductionDependencies (/Users/bpasero/Development/Microsoft/vscode/build/lib/dependencies.js:15:18)
    at Object.getProductionDependencies (/Users/bpasero/Development/Microsoft/vscode/build/lib/dependencies.js:41:20)
    at nodeModules (/Users/bpasero/Development/Microsoft/vscode/build/gulpfile.scan.js:75:38)
    at /Users/bpasero/Development/Microsoft/vscode/build/gulpfile.scan.js:64:3
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (/Users/bpasero/Development/Microsoft/vscode/build/gulpfile.scan.js:32:15) {
  status: 1,
  signal: null,
  output: [
    null,
    '/Users/bpasero/Development/Microsoft/vscode\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@microsoft/1ds-core-js\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@microsoft/1ds-post-js\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@parcel/watcher\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/deviceid\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/iconv-lite-umd\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/policy-watcher\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/proxy-agent\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/ripgrep\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/spdlog\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/sqlite3\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/sudo-prompt\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/tree-sitter-wasm\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/vscode-languagedetection\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/windows-mutex\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/windows-process-tree\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/windows-registry\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/addon-clipboard\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/addon-image\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/addon-search\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/addon-serialize\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/addon-unicode11\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/addon-webgl\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/headless\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@xterm/xterm\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/http-proxy-agent\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/https-proxy-agent\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/jschardet\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/kerberos\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/native-is-elevated\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/native-keymap\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/native-watchdog\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/node-pty\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/open\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/tas-client-umd\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/v8-inspect-profiler\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/vscode-oniguruma\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/vscode-regexpp\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/vscode-textmate\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/windows-foreground-love\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/yauzl\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/yazl\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@microsoft/applicationinsights-core-js\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@microsoft/applicationinsights-shims\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@microsoft/dynamicproto-js\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/is-glob\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/micromatch\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/node-gyp-build\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/fs-extra\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/uuid\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/bindings\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/policy-watcher/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@tootallnate/once\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/agent-base\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/debug\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/socks-proxy-agent\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/proxy-from-env\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/ripgrep/node_modules/yauzl\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/mkdirp\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/spdlog/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/sqlite3/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/tar\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/windows-mutex/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/@vscode/windows-process-tree/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/js-base64\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/kerberos/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/prebuild-install\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/node-pty/node_modules/node-addon-api\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/define-lazy-prop\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/is-docker\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/is-wsl\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/chrome-remote-interface\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/buffer-crc32\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/pend\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/tslib\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/is-extglob\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/braces\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/picomatch\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/graceful-fs\n' +
      '/Users/bpasero/Development/Microsoft/vscode/node_modules/jsonfile\n' [deemon] Build daemon exited.
 *  Terminal will be reused by tasks, press any key to close it. 
 ```

Works fine in a external terminal, so this is something task related, but maybe also npm?

--- a/build/gulpfile.scan.js
+++ b/build/gulpfile.scan.js
@@ -71,12 +71,15 @@ BUILD_TARGETS.forEach(buildTarget => {
 	gulp.task(setupSymbolsTask);
 });
 
-function nodeModules(destinationExe, destinationPdb, platform) {
+function getProductionDependencySources() {
 	const productionDependencies = deps.getProductionDependencies(root);
-	const dependenciesSrc = productionDependencies.map(d => path.relative(root, d)).map(d => [`${d}/**`, `!${d}/**/{test,tests}/**`]).flat();
+	return productionDependencies.map(d => path.relative(root, d)).map(d => [`${d}/**`, `!${d}/**/{test,tests}/**`]).flat();
+}
+
+function nodeModules(destinationExe, destinationPdb, platform) {
 
 	const exe = () => {
-		return gulp.src(dependenciesSrc, { base: '.', dot: true })
+		return gulp.src(getProductionDependencySources(), { base: '.', dot: true })
 			.pipe(filter([
 				'**/*.node',
 				// Exclude these paths.
@@ -89,7 +92,7 @@ function nodeModules(destinationExe, destinationPdb, platform) {
 
 	if (platform === 'win32') {
 		const pdb = () => {
-			return gulp.src(dependenciesSrc, { base: '.', dot: true })
+			return gulp.src(getProductionDependencySources(), { base: '.', dot: true })
 				.pipe(filter(['**/*.pdb']))
 				.pipe(gulp.dest(destinationPdb));
 		};
@@ -99,7 +102,7 @@ function nodeModules(destinationExe, destinationPdb, platform) {
 
 	if (platform === 'linux') {
 		const pdb = () => {
-			return gulp.src(dependenciesSrc, { base: '.', dot: true })
+			return gulp.src(getProductionDependencySources(), { base: '.', dot: true })
 				.pipe(filter(['**/*.sym']))
 				.pipe(gulp.dest(destinationPdb));
 		};

